AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m

```diff
@@ -36,4 +31,18 @@ - (BOOL)isValid {
   return self.type && self.authorityUrl;
 }
 
++ (MSAuthority *)authorityWithDictionary:(NSDictionary *)dictionary {
```

Comment:
```suggestion
+ (instancetype)authorityWithDictionary:(NSDictionary *)dictionary {
```

Reply:
Do you really need this utility method? This will couple the base class with its children.

Reply:
Shouldn't this be named `initWithDictionary` still?

Reply:
I can remove this method and add whole logic to MSAuthConfig if that seems more convenient.